src/order/order_iso_nat.lean

```diff
@@ -43,3 +45,27 @@ theorem well_founded_iff_no_descending_seq [is_strict_order α r] :
     by { rw [function.iterate_succ'], apply h }⟩⟩⟩
 
 end rel_embedding
+
+namespace partial_order
+
+variables (α : Type*) [partial_order α]
+
+/-- For partial orders, one of the many equivalent forms of well-foundedness is the following
+flavour of "ascending chain condition". -/
+def satisfies_acc := ∀ (a : ℕ →ₘ α), ∃ n, ∀ m, n ≤ m → a n = a m
```

Comment:
Should this be an instance?

Reply:
For motivation, I think that there are several classes already where an instance of that class would provide an instance of this, e.g. `[is_noetherian_ring]` would induce this on ideals, and `[wf_dvd_monoid]` would induce this on `associates`.

Reply:
@awainverse Did you mean "Should this be a *class*?"

Reply:
Yes, I did mean "Should this be a class, so that I can provide instances of it?"

Reply:
Good point, done!